### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by thibaultcha**

### CODE REVIEW

File: kong-ngx-build [hunk 1]
```diff
@@ -411,24 +411,28 @@ parse_version() {
   [[ -z $1 ]] && fatal 'missing arg $1 when invoking parse_version()'
   [[ -z $2 ]] && fatal 'missing arg $2 when invoking parse_version()'
 
+  local ver
   local subj=$1
-  local -n ver=$2
   local re='^(-rc[0-9]+$)?[^0-9]*([0-9a-zA-Z]*)(.*?)$'
```
Comment (eskerda) [inline kong-ngx-build]: We could always do a 

```
  [[ ${BASH_VERSION} =~ 3.* ]] && fatal "please upgrade to bash 4 at least"
```

but then we would have gone full circle :D

Comment (thibaultcha) [reply kong-ngx-build]: I just added a commit testing the parsing utilities in Bash 3/4/5.

File: kong-ngx-build [hunk 2]
```diff
@@ -462,8 +466,14 @@ version_lt() {
...
-    [[ "${version_a[index]}" < "${version_b[index]}" ]] && return 0
-    [[ "${version_a[index]}" > "${version_b[index]}" ]] && return 1
+    if [[ ${version_a[index]} =~ ^[0-9]+$ ]]; then
+      [[ "${version_a[index]}" -lt "${version_b[index]}" ]] && return 0
+      [[ "${version_a[index]}" -gt "${version_b[index]}" ]] && return 1
+
+    else
+      [[ "${version_a[index]}" < "${version_b[index]}" ]] && return 0
+      [[ "${version_a[index]}" > "${version_b[index]}" ]] && return 1
+    fi
```
Comment (eskerda) [inline kong-ngx-build]: This would still fail in a comparison like `openssl-1.1.99a openssl-1.1.100a`

File: kong-ngx-build [hunk 3]
```diff
@@ -411,24 +411,28 @@ parse_version() {
...
 
     subj="${BASH_REMATCH[3]}"
     if [[ $subj == "" ]]; then
       break
     fi
   done
+
+  ver="${ver:1}"
+
+  IFS='.' read -r -a $2 <<< "$ver"
```
Comment (eskerda) [inline kong-ngx-build]: I do not really see the point on progressive regex parsing if we end up doing a split by `.` in the end anyway

Comment (thibaultcha) [reply kong-ngx-build]: We get the best of both worlds like so: support for Bash 3, no dependency on Perl and a much faster parsing.

File: kong-ngx-build (hunk 2 above)
Comment (thibaultcha) [reply kong-ngx-build]: Thanks, fixed in the latest version of this branch

**Comment by thibaultcha** [comment]
@eskerda How are we doing with the latest changes in this PR? Seems like all concerns were addressed, is it working in your environment now?

**Pull request merged**